### ISSUE TITLE
magento generate a broken js-translation.json file while deploying locale 

### ISSUE THREAD

**Issue by sezio** [magento/magento2#18519]
### Preconditions

1. magento CE 2.2.6 or 2.2.5 with porto theme
2. apache 2.2
3. php 7.0.30 
4. data were migrated from magento 1 using the migration tool 

### Steps to reproduce

I have no idea of how it could be reproduced :disappointed: 
However, context is the following : I've been migrating some Magento 1 data using the migration tool on a clean magento 2.2.5 (and then on 2.2.6 hoping the problem would be gone). 
Deploying the static content with `bin/magento setup:static-content:deploy -f fr_FR` creates an invalid js-translation.json file

### Expected result

A fine js-translation.json containing valid json as : 
`{"Save":"Sauvegarder","Close":"Fermer","Submit":"Envoyer",...}`

### Actual result

A broken js-translation.json file containing non replaced string with triple curved brackets : 
`{"Method %s does not exist on jQuery.decorate":"{{{Method %s does not exist on jQuery.decorate}}{{Method %s does not exist on jQuery.decorate}}{{Method %s does not exist on jQuery.decorate}}{{themeSmartwave\/porto}}}","Please wait...":"{{{Please wait...}}{{Please wait...}}{{Please wait...}}{{themeSmartwave\/porto}}}",`

### Additional information 
1. No error in the logs or thrown by the deployment, file is generated but content is broken 
2. I have the problem on my staging env but not on my local env... Both have the same php version running (7.0.30) on debian stretch and I use apache2 with the same modules enabled : `php -i` and `apachectl -M` return the exact same values between the two environments 
3. on my staging env, js-translation.json is broken in each deployed theme, even the ones from Magento 
4. Same problem on every locales 

I've also found the following issue #16633 which is the exact the same problem but was closed because it needed an update

Thanks for your help !

**Comment by magento-engcom-team** [comment]
Hi @sezio. Thank you for your report. 
To help us process this issue please make sure that you provided the following information:
- [ ] Summary of the issue
- [ ] Information on your environment
- [ ] Steps to reproduce
- [ ] Expected and actual results

Please make sure that the issue is reproducible on the vanilla Magento instance following **Steps to reproduce**. To deploy vanilla Magento instance on our environment, please, add a comment to the issue: 

`@magento-engcom-team give me $VERSION instance` 

where `$VERSION` is [version tags](https://github.com/magento/magento2/tags) (starting from 2.2.0+) or develop branches (for example: 2.3-develop). 
For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation.

@sezio do you confirm that you was able to reproduce the issue on vanilla Magento instance following steps to reproduce?
- [ ] yes
- [ ] no

**Comment by sezio** [comment]
After starting again and again my migration, I finally have a clue of what is the problem... During the settings migration, some rows for inline translation are imported in the core_config_data table; it looks like you have to delete them to have a clean json file... 
Here are the sql queries I ran : 

```
delete from core_config_data where path like 'dev/translate_inline/active';
delete from core_config_data where path like 'dev/translate_inline/active_admin';
```

I really hope it will be useful for anyone else as I was about to become crazy.... 

But I still don't get why it was working fine on my development env and it was not on my staging host... Both have the same config!

**Comment by magento-engcom-team** [comment]
Hi @engcom-backlog-nazar. Thank you for working on this issue. 
In order to make sure that issue has enough information and ready for development, please read and check the following instruction: :point_down:
- [ ] 1. Verify that issue has all the required information. (**Preconditions**, **Steps to reproduce**, **Expected result**, **Actual result**).<details><summary>Details</summary>If the issue has a valid description, the label `Issue: Format is valid` will be added to the issue automatically. Please, edit issue description if needed, until label `Issue: Format is valid` appears.</details>

- [ ] 2. Verify that issue has a meaningful description and provides enough information to reproduce the issue. If the report is valid, add `Issue: Clear Description` label to the issue by yourself.

- [ ] 3. Add `Component: XXXXX` label(s) to the ticket, indicating the components it may be related to.
- [ ] 4. Verify that the issue is reproducible on `2.3-develop` branch<details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.3-develop instance` to deploy test instance on Magento infrastructure. <br/> - If the issue is reproducible on `2.3-develop` branch, please, add the label `Reproduced on 2.3.x`.<br/>- If the issue is not reproducible, add your comment that issue is not reproducible and close the issue and **_stop verification process here_**!
</details>

- [ ] 5. Verify that the issue is reproducible on `2.2-develop` branch. <details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.2-develop instance` to deploy test instance on Magento infrastructure. <br/>- If the issue is reproducible on `2.2-develop` branch, please add the label `Reproduced on 2.2.x`</details>

- [ ] 6. Add label `Issue: Confirmed` once verification is complete. 

- [ ] 7. Make sure that automatic system confirms that report has been added to the backlog.

**Comment by ghost** [comment]
Hi @sezio   thank you for your report,  problems with data migration from 1.x must be reported in https://github.com/magento/data-migration-tool repository.

Best Regards

**Comment by hostep** [comment]
Reopening for further investigation. This has nothing to do with data migration btw.

We ran against the same problem, because for some reason the configuration `dev/translate_inline/active` was activated on a production shop of ours (but not on staging of course ...).
We just finished upgrading the shop from 2.1.12 to 2.1.16 and deployed it to production today, and it broke the `js-translation.json` file just like reported above.

Disabling the Inline translation setting 'Enabled for Storefront' and performing a redeploy fixed it.

I'm reopening, so we can verify if this issue is still happening on 2.2 or 2.3.

So as far as I can tell, it should be reproducible by:
- enabling the inline translations
- running `bin/magento setup:static-content:deploy some/theme en_US`
- checking the file `pub/static/frontend/some/theme/en_US/js-translation.json` and seeing if translations are encapsulated by `{{` & `}}` characters.

**Comment by hostep** [comment]
Confirmed on latest 2.3-develop branch (used commit 5791131b556)

### Steps to reproduce
- Setup vanilla Magento installation
- Run `bin/magento config:set dev/translate_inline/active 1` (activates the inline translation feature)
- Run `bin/magento config:set dev/restrict/allow_ips 1.2.3.4` (use an IP which is not your own, so you don't see the inline translation active on the frontend)
- Put Magento in production mode: `bin/magento deploy:mode:set production`

### Expected result
- Generated `js-translation.json` file contains correct translations
- No broken translations in frontend:

<img width="437" alt="Screenshot 2019-04-07 at 09 29 06" src="https://user-images.githubusercontent.com/85479/55680261-b0fcfb00-5917-11e9-8d26-4a85325854a3.png">

- No broken translations in backend:

<img width="1793" alt="Screenshot 2019-04-07 at 09 29 27" src="https://user-images.githubusercontent.com/85479/55680263-b5c1af00-5917-11e9-9a8f-a6c599ff3d02.png">

### Actual result
- Generated `js-translation.json` file contains incorrect translations
- Broken translations in frontend:

<img width="472" alt="Screenshot 2019-04-07 at 09 25 25" src="https://user-images.githubusercontent.com/85479/55680230-3633e000-5917-11e9-8166-1903325a6e74.png">

- Broken translations in backend

<img width="1477" alt="Screenshot 2019-04-07 at 09 25 43" src="https://user-images.githubusercontent.com/85479/55680235-3e8c1b00-5917-11e9-8777-95c485496066.png">



Thanks to @sezio for finding the cause of this problem!

**Comment by sezio** [comment]
Hi @hostep 

Indeed, it has nothing to do with data migration, your summary of the issue is very clear and that's exactly the problem. 
Thanks for completing the issue!

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @engcom-backlog-nazar 
 Thank you for verifying the issue. Based on the provided information internal tickets `MAGETWO-99356` were created

**Issue Available**: @engcom-backlog-nazar, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._

**Comment by magento-engcom-team** [comment]
Hi @sezio. 

Thank you for your report and collaboration!

The related internal Jira ticket `MC-24336` was closed as `Fixed`. 

The fix will be available with the upcoming `2.4.3` release.

**Comment by hostep** [comment]
Comment above seems to be incorrect. But it appears that this problem is fixed anyways in Magento `2.4-develop` at least after quickly testing it.

If I have to guess, it might have been this PR that fixed it: https://github.com/magento/magento2/pull/28856, which was included in Magento 2.4.1
